### PR TITLE
Components: remove `config()` utility function, use values directly

### PR DESCRIPTION
**Pull request by ciampo**
## Description

Remove the (redundant) `config()` utility function, and instead access the config values directly. (part of what highlighted in #30503)

## How has this been tested?

- [ ] Make sure that the project builds correctly
- [ ] Make sure that the Spinner and Input components look and behave like on `trunk` (e.g. by using [Storybook](https://github.com/WordPress/gutenberg/blob/trunk/storybook/README.md))

## Screenshots 

N/A

## Types of changes

Refactor / cleanup

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. 
- [x] My code follows the accessibility standards. 
- [ ] N/A I've tested my changes with keyboard and screen readers. 
- [x] My code has proper inline documentation. 
- [ ] N/A I've included developer documentation if appropriate. 
- [ ] N/A I've updated all React Native files affected by any refactorings/renamings in this PR (please manually search all `*.native.js` files for terms that need renaming or removal).

### CODE REVIEW

File: packages/components/CHANGELOG.md [hunk 1]
```diff
@@ -15,6 +15,7 @@
 ### Internal
 
 -   `Flex`, `FlexBlock`, and `FlexItem` components have been re-written from the ground up ([#31297](https://github.com/WordPress/gutenberg/pull/31297)).
+-   `config()` utility function has been removed, in favour of accessing config values directly ([#31646](https://github.com/WordPress/gutenberg/pull/31646))
```
Comment (diegohaz) [inline packages/components/CHANGELOG.md]: I don't think we should add an entry here as it's not even touching any user-facing API.

**Review by diegohaz — APPROVED** [review]
This looks great, thanks @ciampo!
I made a comment about the changelog entry, but it's not a blocker.

File: packages/components/CHANGELOG.md (hunk 1 above)
Comment (ciampo) [reply packages/components/CHANGELOG.md]: Makes sense! I deleted the entry in https://github.com/WordPress/gutenberg/pull/31646/commits/f6735f3a45dbac7fded13736b50f028198e5e474

**Comment by ciampo** [comment]
Hey @diegohaz , thank you for your review! I also went ahead and rebased to solve some conflicts.

Since I don't have repo privileges yet, would you mind merging the PR? Thank you!

**Pull request merged**